common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDB.java

```diff
@@ -457,4 +480,19 @@ byte[] getPrefix(RocksDBTypeInfo.Index idx) throws Exception {
 
   }
 
+  private static class RocksDBIteratorWeakReference extends WeakReference<RocksDBIterator<?>> {
```

Comment:
`RocksDBIteratorWeakReference` use to hold `RocksIterator`,  better as the inner class of `RocksDB` or `RocksDBIterator`? I want your suggestion @dongjoon-hyun

Reply:
[65e0c00](https://github.com/apache/spark/pull/36467/commits/65e0c00077906ce8a2ff0640278395a7b4e3f914) move `RocksDBIteratorWeakReference` to `RocksDBIterator` first to avoid  add `Getter` of `o.rocksdb.RocksIterator`